bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/LinkyAuthServlet.java

```diff
@@ -0,0 +1,176 @@
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.util.MultiMap;
+import org.eclipse.jetty.util.StringUtil;
+import org.eclipse.jetty.util.UrlEncoded;
+import org.openhab.binding.linky.internal.handler.ApiBridgeHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The LinkyAuthServlet manages the authorization with the Linky Web API. The servlet implements the
+ * Authorization Code flow and saves the resulting refreshToken with the bridge.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ * @author Laurent Arnal - Rewrite addon to use official dataconect API
+ */
+@NonNullByDefault
+public class LinkyAuthServlet extends HttpServlet {
+
+    private static final long serialVersionUID = -4719613645562518231L;
+
+    private static final Pattern MESSAGE_KEY_PATTERN = Pattern.compile("\\$\\{([^\\}]+)\\}");
+
+    private static final String CONTENT_TYPE = "text/html;charset=UTF-8";
+
+    private static final String HTML_USER_AUTHORIZED = "<p class='block authorized'>Addon authorized for %s.</p>";
+    private static final String HTML_ERROR = "<p class='block error'>Call to Enedis failed with error: %s</p>";
+
+    // Keys present in the index.html
+    private static final String KEY_AUTHORIZE_URI = "authorize.uri";
+    private static final String KEY_RETRIEVE_TOKEN_URI = "retrieveToken.uri";
+    private static final String KEY_REDIRECT_URI = "redirectUri";
+    private static final String KEY_PRMID_OPTION = "prmId.Option";
+    private static final String KEY_AUTHORIZED_USER = "authorizedUser";
+    private static final String KEY_ERROR = "error";
+    private static final String KEY_PAGE_REFRESH = "pageRefresh";
+
+    private final Logger logger = LoggerFactory.getLogger(LinkyAuthServlet.class);
+    private final String indexTemplate;
+
+    private ApiBridgeHandler apiBridgeHandler;
+
+    public LinkyAuthServlet(ApiBridgeHandler apiBridgeHandler, String indexTemplate) {
+        this.indexTemplate = indexTemplate;
+        this.apiBridgeHandler = apiBridgeHandler;
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        logger.debug("Linky auth callback servlet received GET request {}.", req.getRequestURI());
+        final Map<String, String> replaceMap = new HashMap<>();
+
+        StringBuffer requestUrl = req.getRequestURL();
+        String servletBaseUrl = requestUrl != null ? requestUrl.toString() : "";
+
+        String servletBaseURLSecure = servletBaseUrl;
```

Comment:
`servletBaseUrl` does not seem used after this.

Reply:
I refactor this, removing duplicates var